store/sqlstore/plugin_store.go

```diff
@@ -225,6 +234,19 @@ func (ps SqlPluginStore) DeleteAllForPlugin(pluginId string) *model.AppError {
 	return nil
 }
 
+func (ps SqlPluginStore) deleteIfExpired(pluginId, key string) *model.AppError {
+	currentTime := model.GetMillis()
+	if _, err := ps.GetMaster().Exec("DELETE FROM PluginKeyValueStore WHERE PluginId = :PluginId AND PKey = :Key AND ExpireAt != 0 AND ExpireAt < :CurrentTime",
+		map[string]interface{}{
+			"PluginId":    pluginId,
+			"Key":         key,
+			"CurrentTime": currentTime,
+		}); err != nil {
+		return model.NewAppError("SqlPluginStore.Delete", "store.sql_plugin_store.delete.app_error", nil, fmt.Sprintf("plugin_id=%v, key=%v, err=%v", pluginId, key, err.Error()), http.StatusInternalServerError)
```

Comment:
If we don't inline, let's change the error text to be `SqlPluginStore.deleteIfExpired`,

Reply:
Personally, I'd prefer this. I'd argue it's more clear to see and understand the intentions of deleteIfExpired over another inline SQL statement. I'm open to inlining this if you feel strongly about it though.